src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -123,13 +123,27 @@ void OTARequestor::OnQueryImageResponse(void * context, const QueryImageResponse
             return;
         }
 
-        MutableByteSpan updateToken(requestorCore->mUpdateTokenBuffer);
-        CopySpanToMutableSpan(update.updateToken, updateToken);
-        requestorCore->mTargetVersion = update.softwareVersion;
-        requestorCore->mUpdateToken   = updateToken;
+        if (update.softwareVersion > requestorCore->mCurrentVersion)
+        {
+            ChipLogDetail(SoftwareUpdate, "Update available from %" PRIu32 " to %" PRIu32 " version",
+                          requestorCore->mCurrentVersion, update.softwareVersion);
+            MutableByteSpan updateToken(requestorCore->mUpdateTokenBuffer);
+            CopySpanToMutableSpan(update.updateToken, updateToken);
```

Comment:
This can fail, if the size of thw destination is not big enough.  Can that happen here?

Reply:
Theoretically it can if the provider somehow didn't follow the spec. Maybe the right thing to do is perform a length check of updateToken in `ExtractUpdateDescription`?

Reply:
Done